### DISCUSSION TITLE
Value relation issue

### DISCUSSION THREAD

**Discussion by dragonfire151** [opengisch/QField#2983] [Q&A]
Ever since I updated both QField and the Android Samsung tablet I've been having a value relation issue.

I have two layers: Station, Structure
The Station layer has a Field Station ID. I also like to record the Station ID in the Structure layer.

Previously I had this setup in the Structure Layer Attribute Form and it would work in QField:
![image](https://user-images.githubusercontent.com/107634973/174103314-1f2cc452-d893-4008-ba67-2d64b1cdd0ff.png)

But since the updates, I am unable to record a Station ID in the Station layer in QField. For some reason it is forcing a drop down (can't text edit) that is completely empty.

When I test everything in QGIS it works with no issue.

I should probably set this up properly with a relation but this was a nice quick set up that, until now, worked.

EDIT: Stripped down project added
[QField relation test.zip](https://github.com/opengisch/QField/files/8920885/QField.relation.test.zip)

I'm using a Samsung tablet, Android version 12. QField version 2.1.4 Bumblebee

**Comment by nirvn** [comment]
Feel free to attach a test project, that'll increase chances of people looking at this.

Also, have you tried enabling the use completer checkbox?

**Reply by dragonfire151** [reply]
Added a stripped down project to the initial post. 

I made some attempts with completer but ran into other issues.

**Comment by nirvn** [comment]
https://user-images.githubusercontent.com/1728657/174199248-1eaaf2a1-4bc2-4a71-8e06-8fab881157c5.mp4

This is what I get when trying your test project. I can see pre existing stations in the drop-down combobox and I can search for specific stations if I click on the magnification glass next to the combobox. I can't see what's broken in there, am I misunderstanding you?

**Reply by dragonfire151** [reply]
The problem is when you try to add a new station in the station layer. The structure layer works as desired.

**Comment by nirvn** [comment]
@dragonfire151 , can you attach a screencast please?

**Comment by dragonfire151** [comment]
When I bring up station and try to add a new one it shows a drop down with no data. Previously I was able to edit this field by adding text.

![Screenshot_20220617-073113](https://user-images.githubusercontent.com/107634973/174300251-75b23e23-4a44-453a-be96-6e7047bd578a.jpg)

**Reply by dragonfire151** [reply]
I'm using a Samsung tablet, Android version 12. QField version 2.1.4 Bumblebee

**Reply by nirvn** [reply]
@dragonfire151 , can you test the following beta APK: https://github.com/opengisch/QField/pull/2963#issuecomment-1158743790

I think the problem has already been addressed during the 2.2 dev cycle.

**Reply by dragonfire151** [reply]
I tried out that download and still the same issue is occurring.

**Reply by nirvn** [reply]
@dragonfire151 , this is what I see:


https://user-images.githubusercontent.com/1728657/174311590-e8c534c9-d29f-4fce-bd85-07f1f408c0d0.mp4

**Reply by dragonfire151** [reply]
Yea that's how it should be working. Oof ok. The last thing to update would be the version of QGIS I'm running and that requires getting IT involved because my workplace is ridiculous. 

Thank you so much for your help!

**Reply by nirvn** [reply]
@dragonfire151 , I'm using the zipped project you attached above against latest QField dev. Unless you did _not_ install the right APK, you should see the same Zip you attached working for you too. I don't think you need to update QGIS. 

Can you take a screenshot of the About QField panel (from the APK build you installed) and make a video of your issue? I'd like to rule out any lingering regression.

**Reply by dragonfire151** [reply]
https://user-images.githubusercontent.com/107634973/174320339-0c4b2182-4ee2-
![Screenshot_20220617-094023](https://user-images.githubusercontent.com/107634973/174320551-bbe041c7-a4e7-419a-a63c-4d99f0c9cb21.jpg)
4f90-a842-a1f7faea84da.mp4

Hope these help!

**Reply by nirvn** [reply]
Well, you definitely got the right build, and the field is visibly not configured properly. I'm perplexed by this. What's your device model?

**Reply by dragonfire151** [reply]
It's a Samsung Galaxy Tab Active3

I too am very perplexed! We don't use the devices through the winter so when gearing up for field work everything needed an update, the tablet, QField, and the QField plugin within QGIS. This caused a few headaches, like the new file location on the tablet for data transfer. But this is the last bug to fix. 

I have a work around and have simply disabled entering the station information on the structure layer and changed the Station ID field on the Station layer to a simple text edit. But when presented with a problem like this it's hard to not want to solve it.

**Reply by nirvn** [reply]
@dragonfire151 , when you were testing just now, were you using the same zipped project you attached to this discussion or something slightly different? One of the nice things you can do with QField 2.1 is downloading the zip file in your browser and open it with QField, it'll auto import the project. Maybe try that here to rule out a slightly different project on your end?

**Reply by dragonfire151** [reply]
Wow. That worked.

Previously I had used the QField plugin in QGIS to make an exported project. I then copied that exported project folder to the tablet and that one was causing me issues. So the exporter in the plugin update must be off somehow. That makes sense since an older project I had exported via the QField plugin before the plugin update worked fine.

So I guess I'll just zip everything from now on!

Thanks again for going through this process with me!

**Reply by jpnunn** [reply]
H. I am getting the same results using qfield 2.4.5 on a pixel 4a (and also samsung tablets).   The first layer in QGIS that has a field set to "text edit" appears as expected in QGIS which allows for adding features but as a combo box in qfield without the ability to add new featuers.  It only does this if other layers are set as value relations linking back to this layer and field.    I have imported the project via a zip file and get these results.

**Reply by jpnunn** [reply]
Slight update - still not fixed though!  I changed the field names so they were NOT the same as i thought this might be the cause.  Now in QGIS it still work as expected (one text edit, one combobox), but in QFIELD both the fields in both layers are just text edit input, despite one being setup as a value relation field (so should be combobox).

**Reply by jpnunn** [reply]
Fixed!  - the problem was the fieldnames were identical in both layers.  I altered one of the fieldnames to be slightly different to the other and it worked fine. Text entry in one layer and Drop down in another as expected.

**Reply by dragonfire151** [reply]
Glad you got it figured out! I was thinking on it yesterday but wasn’t sure.

**Reply by nirvn** [reply]
It's really nice to see the community stepping up and helping each other. Keep it going 💚
